engine/src/main/java/org/terasology/rendering/dag/nodes/FinalPostProcessingNode.java

```diff
@@ -95,15 +95,17 @@ public void process() {
     }
 
     @Override
-    public void propertyChange(PropertyChangeEvent evt) {
-        // we assume here that this property change event is fired only if there has been a change
-        if (!renderingDebugConfig.isEnabled()) {
-            removeDesiredStateChange(enableDebugMaterial);
-            addDesiredStateChange(enablePostMaterial);
-        } else {
-            removeDesiredStateChange(enablePostMaterial);
-            addDesiredStateChange(enableDebugMaterial);
+    public void propertyChange(PropertyChangeEvent event) {
+        if (event.getOldValue() != event.getNewValue()) {
```

Comment:
See comment above regarding event being fired only if the value is indeed changed.